### DISCUSSION TITLE
Making the "-all" versions available in maven repository

### DISCUSSION THREAD

**Discussion by shazzerim** [checkstyle/checkstyle#12693] [Ideas]
Hi there!

We're using the "checkstyle-*-all.jar" artifact, which can be downloaded in github exclusively. 
Is there a reason this version isn't available as separate classifier in any official maven repository? Because it would make our life so much easier if it where and we could let maven and renovate handle the dependencies in our projects.

**Comment by rnveach** [comment]
Maven doesn't store "all" jars. This is because an "all" jar is just the main dependency and any sub-dependencies it uses. When you naturally build a maven project, it automatically combines all dependencies at runtime into what amounts as the "all" jar. If maven was used to store "all" jars, then it would be duplicating storage as mentioned and cause them a huge waste of space.

If you are using maven for your process or something that hooks into maven, you just need to include Checkstyle as a dependency and maven will do the rest including the sub-dependencies.
https://github.com/sevntu-checkstyle/checkstyle-samples/blob/master/maven-project/pom.xml#L33-L37

If you are not using maven, then the "all" jar hosted at GH is the preferred way to download it.
